tests/test_dataflow/core/test_linearize.py

```diff
@@ -97,31 +96,22 @@ def test_unnest_line():
 def test_sexp_formatter_and_deformatter():
     """Round-trip tests for s-expression formatter and deformatter."""
     data = [
-        ("", ""),
-        ('# ( String "singleToken" )', '# ( String " singleToken " )'),
-        ('# ( String "multiple tokens" )', '# (  String " multiple tokens " )'),
-        # these two are not valid value s-expressions, so they are not formatted
-        ('# ( String "open quote only )', '# (  String "open quote only )'),
-        ('# ( String closing quote only" )', '# ( String closing quote only" )',),
-        # multiple value s-expressions
-        (
-            '# ( String "singleToken" ) # ( String "multiple tokens" )',
-            '# ( String " singleToken " ) # ( String " multiple tokens " )',
-        ),
+        ('#(String "singleToken")', '# ( String " singleToken " )'),
+        ('#(String "multiple tokens")', '# (  String " multiple tokens " )'),
         # real data
         (
-            '( ( mapGet ( # ( Path "fare.fare_id" ) ) ( clobberRevise ( getSalient ( actionIntensionConstraint ) ) ( '
-            "Constraint[Constraint[flight]] ) ( Constraint "
-            ':type ( ?= ( # ( String "fare" ) ) ) ) ) ) )',
-            '( ( mapGet ( # ( Path " fare.fare_id " ) ) ( clobberRevise ( getSalient ( actionIntensionConstraint ) ) '
+            '((mapGet #(Path "fare.fare_id") (clobberRevise (getSalient (actionIntensionConstraint)) ('
+            "Constraint[Constraint[flight]]) (Constraint "
+            ':type (?= #(String "fare"))))))',
+            '( ( mapGet # ( Path " fare.fare_id " ) ( clobberRevise ( getSalient ( actionIntensionConstraint ) ) '
             "( Constraint[Constraint[flight]] ) ( "
-            'Constraint :type ( ?= ( # ( String " fare " ) ) ) ) ) ) )',
+            'Constraint :type ( ?= # ( String " fare " ) ) ) ) ) )',
         ),
     ]
 
     for raw_sexp, formatted_sexp in data:
-        assert sexp_formatter(raw_sexp.split()) == formatted_sexp.split()
-        assert sexp_deformatter(formatted_sexp.split()) == raw_sexp.split()
+        assert lispress_to_seq(parse_lispress(raw_sexp)) == formatted_sexp.split()
```

Comment:
sorry I'm a little confused now. This test `test_sexp_formatter_and_deformatter` specifically checks the `sexp_formatter` and `sexp_deformatter`. Why we remove the `sexp_deformatter` call now?

Reply:
I think we were testing implementation details. Do you think it's important to test the formatter itself, or just that roundtripping works and that formatting works as expected? I can rename the test.

Reply:
Yeah, I think it's better to rename the test. Thanks!

Reply:
Done